src/docfx/build/toc/ServicePageGenerator.cs

```diff
@@ -72,8 +72,20 @@ internal class ServicePageGenerator
                     {
                         if (topLevelTOCRelativeDir != null)
                         {
+                            string? hrefFileFullPath;
+
                             var topLevelTOCYmlDir = Path.GetFullPath(Path.Combine(_docsetPath, topLevelTOCRelativeDir));
-                            var hrefFileFullPath = Path.GetFullPath(Path.Combine(topLevelTOCYmlDir == null ? "" : topLevelTOCYmlDir, childHref));
+
+                            if (childHref.StartsWith("~/") || childHref.StartsWith("~\\"))
```

Comment:
Is there a reason why service pages are generated before resolving TOC links? If service pages are generated _after_ TOC link resolve, then presumably we don't have this problem?

Reply:
Service page yml files are generated before resolving. 
But if we don't do this, it will get wrong results. For example, "href: ~/api/doc.md" will be resolved as "../../~/api/doc.md" and result in an invalid file link error.